### ISSUE TITLE
Switching "Read in English" on doesn't switch the language on the navigation bars and page title

### ISSUE THREAD

**Issue by andre-ss6** [MicrosoftDocs/feedback#1721]
**Describe the bug**
When switching "Read in english" on, the page main content is switched to english, but the side navigation bars and the page title on the browser tab remain translated.

**To Reproduce**
1. Go to https://docs.microsoft.com/pt-br/aspnet/core/web-api/advanced/formatting?view=aspnetcore-2.2
2. Click on "Ler em Inglês"
3. See error

**Expected behavior**
Everything is switched to english

**Screenshots**
![image](https://user-images.githubusercontent.com/15827357/61137797-90a38600-a49c-11e9-8300-8cc0221ab24a.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 1903
 - Browser: Google Chrome
 - Version: 75

**Comment by ryanmajidi** [comment]
@andre-ss6 this is by design. To change your locale click the link in the bottom left corner.
![image](https://user-images.githubusercontent.com/7329992/61669628-9451bc80-ac95-11e9-9568-e40ab161f511.png)

**Comment by andre-ss6** [comment]
@ryanmajidi Oh... Geez. I don't even....... Cmon. What happens inside Microsoft? You must all be drugged. 

There have been tons of issues opened over the years covering the issue of having to read translated docs (and badly translated ones, it's important to note), from Connect, to UserVoice to here, on Github. I believe this is the _very_ reason the "Read in English" switch exists. If you believe that the link to change the locale is enough, then please, explain to me, _why is the "Read in English" switch necessary_? And why in the earth would this be _by design_? If the user is _explicitly_ telling you that not only he can indeed read English, but also prefer to do so, then why keep the navigation stuff localized?!

So... sigh... The issue is that when you come from Google, the docs come translated. I use Google on both PT-BR _and_ EN-US, however Microsoft docs always show up on the former. Switching the locale as you suggested doesn't help. Every time I reach the docs from a link from Google, the locale specified at the URL is the one honored, instead of the one I chose before.

As many people know (and have reported), reading translated documentation, even when you're not proficient in English, is not a great experience, because you end up getting translated exception messages, for example, and other technical messages, which are far better served in English (one of the major advantages being that you can easily find help online, and translating localized technical messages back to English is not an easy task). And then you also have to remember that automatic translation is really, _really_ bad. Most of the time it does more harm than it does good. I mean, it really is _bad_. And ok, at least now the docs are open source, so anyone can help translate. But then you have the final problem: they won't be always up to date.

In short: Translated docs sucks and there have been lots of reports about this over the years; Microsoft docs always appear localized on Google; and, finally, this very issue: the switch, which is the alternative right now, isn't working 100%.

**Comment by mairaw** [comment]
@SonjaSaltzman FYI - this is not the first time I see users requesting that the whole chrome would just switch to the English version instead of just the article. Is that something we can consider?

**Comment by SonjaSaltzman** [comment]
@bodhigautam could you please explain current functionality of "Read in English" toggle versus "Language Picker", and also our plans to implement a feature that would address the user's issue with Google Search?

**Comment by bodhigautam** [comment]
@andre-ss6 The read in English toggle activation shows the content in English corresponding to what has been localized. The English content is already loaded and hidden in the backend. On activating the toggle, the hidden English content is shown and the localized content hides back. The side navigation bars don't support this property as only the conceptual content contains localized and English content next to each other. Hence you will see there is no change in url, neither a new page getting loaded when the toggle is enabled.
If you want to see all the content in English, you can select the language from the Language Picker on the bottom left and select your language as English. This will reload the page in the English Language along with the navigation and page title. 
We are working on a solution to understand customer's preferred language and giving an option to the user to always navigate to his preferred language page if he doesn't land there from search. In this way, the user will always be redirected from other language page to his/her preferred language page.
So if you select English as your preferred language in the Docs using Language picker. The next time you land on a non English page, we will ask if you always want to get redirected to your language page. If you select yes, we will ensure that the page is redirected to your preferred locale always (in your case English).
cc @SonjaSaltzman @mairaw

**Comment by andre-ss6** [comment]
> We are working on a solution to understand customer's preferred language and giving an option to the user to always navigate to his preferred language page if he doesn't land there from search. In this way, the user will always be redirected from other language page to his/her preferred language page.
So if you select English as your preferred language in the Docs using Language picker. The next time you land on a non English page, we will ask if you always want to get redirected to your language page. If you select yes, we will ensure that the page is redirected to your preferred locale always (in your case English).

Yes, this would be great.

**Comment by andre-ss6** [comment]
@SonjaSaltzman @mairaw @bodhigautam Hey, is there an issue to track that aforementioned solution? Is it still planned at all?

**Comment by andre-ss6** [comment]
This is still an issue :(

**Comment by bodhigautam** [comment]
@andre-ss6 This solution is still not in place. Its in our backlog state currently and we will soon start developing it. Apologies for the delay in delivering this solution. Will keep you posted on the redirection feature.
@dend This feedback is about the locale redirection feature. 
@SonjaSaltzman @mairaw JFYI.
